app/views/providers/allocations/pick_a_provider.html.erb

```diff
@@ -14,8 +14,18 @@
     <ul class="govuk-list govuk-list--bullet">
       <% training_providers.each do |training_provider| %>
         <li>
-          <%= govuk_link_to "#{training_provider.provider_name} (#{training_provider.provider_code})",
-            initial_request_provider_recruitment_cycle_allocations_path(training_provider_code: training_provider.provider_code) %>
+          <% allocation = Allocation.for_provider_and_training_provider(recruitment_cycle: @recruitment_cycle, provider: @provider, training_provider: training_provider) %>
```

Comment:
Is there any particular reason why we are running an active record query from the view, rather than returning the data from the controller?

Reply:
- theres no ActiveRecord here its the json api
- the api as it stands currently does not allow for loading of allocations along with training providers which is why unfortunately it is done here in the view

Reply:
Yeah sorry I mean json api 😄

Reply:
Ok thanks for the explanation